File: c4/cpp/c4.cpp
#include <iostream>

int main()
{
    int pancakes[10];
    int person[10];
    char input[10];
    int Iinput;
    bool hasSwapped = true;

    //Let's get the info on the pancakes!
    for (int i = 0; i < 10; i++)
    {
	pancakes[i] = -1;
	person[i] = i;

	while (pancakes[i] == -1){
	    try
	    {
		std::cout << "How many pancakes did Person " << i + 1 << " eat?: ";
		std::cin >> input;
		Iinput = std::stoi(input);
		pancakes[i] = Iinput;
	    } catch (const std::invalid_argument e)
	    {
		std::cout << "That's invalid" << std::endl;
		pancakes[i] = -1;
	    }
	}

    }

    //Let's sort out some pancakes! (bubble sort)
    while (hasSwapped)
    {
	hasSwapped = false;
	for(int i = 1; i < 10; i++)
	{
	    if (pancakes[i-1] < pancakes[i])
	    {
		int tempPancakes = pancakes[i-1];
		int tempPerson = person[i-1];

		pancakes[i-1] = pancakes[i]; //Swap the number of pancakes
		pancakes[i] = tempPancakes;

		person[i-1] = person[i]; //Swap the corresponding person
		person[i] = tempPerson;

		hasSwapped = true;
	    }
	}
    }

    for(int i = 0; i < 10; i++)
    {
	if (pancakes[i] == 1)
	{
	    std::cout << "Person " << person[i] + 1 << ": " << pancakes[i] << " pancake" << std::endl;
	} else {
	    std::cout << "Person " << person[i] + 1 << ": " << pancakes[i] << " pancakes" << std::endl;
	}
    }
    
    std::cout << "Person " << person[0] + 1 << " ate the most pancakes." << std::endl;
    std::cout << "Person " << person[9] + 1 << " ate the least pancakes." << std::endl;

    return 0;
}

File: c4/python/c4.py
people = []
pancakes = []

#Input for number of pancakes
for i in range(10):
    people.append(i + 1)
    pancakes.append(-1)
    while pancakes[i] == -1:
        try:
            pancakes[i] = int(input("How many pancakes did Person {} eat?:".format(i + 1)))
        except:
            print("That's not valid")
            pancakes[i] = -1

#Sort out the pancakes
hasSwapped = True
while hasSwapped:
    hasSwapped = False
    for i in range(1, len(pancakes)):
        if (pancakes[i-1] < pancakes[i]):
            pancakes[i-1], pancakes[i] = pancakes[i], pancakes[i-1]
            people[i-1], people[i] = people[i], people[i-1]
            hasSwapped = True;

#Output the results
print()
for i in range(len(pancakes)):
    if pancakes[i] == 1:
        print("Person %d: %d pancake" % (people[i], pancakes[i]))
    else:
        print("Person %d: %d pancakes" % (people[i], pancakes[i]))
print("Person {} ate the most pancakes".format(people[0]))
print("Person {} ate the least pancakes".format(people[len(people) - 1]))
